include/xtensor/xrandom.hpp

```diff
@@ -0,0 +1,195 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+/**
+ * @brief functions to obtain xgenerators generating random numbers with given shape
+ */
+
+#ifndef XRANDOM_HPP
+#define XRANDOM_HPP
+
+#include <utility>
+#include <random>
+#include <memory>
+#include <sstream>
+
+#include "xgenerator.hpp"
+
+namespace xt
+{
+    namespace detail
+    {
+        template <class D, class S, class V, class E = std::mt19937>
+        struct random_impl
+        {
+            using value_type = V;
+            using size_type = std::size_t;
```

Comment:
Just a styling thing: I would add `using` statements for `distribution_type`, `strides_type`, `engine_type`.